Sources/Datadog/DatadogCore/Utils/FoundationExtensions.swift

```diff
@@ -0,0 +1,28 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+
+extension DispatchQueue {
+    /// Flushes the queue by dispatching a chain of asynchronous operations (one after
+    /// another) and awaiting completion of the last one.
+    ///
+    /// - Parameter numberOfTimes: number of `async {}` operations to chain
+    func flush(numberOfTimes: Int) {
+        let semaphore = DispatchSemaphore(value: 0)
+
+        func recursiveFlush(countDown: Int) {
+            if countDown > 0 {
+                self.async { recursiveFlush(countDown: countDown - 1) }
+            } else {
+                semaphore.signal()
+            }
+        }
+
+        recursiveFlush(countDown: numberOfTimes)
+        _ = semaphore.wait(timeout: .distantFuture)
+    }
```

Comment:
Maybe it could be moved to tests in that case?

Reply:
This looks odd to me TBH, especially if it's not yet necessary. Maybe it worth adding this when we need it, no?
It's might be that our threading model is at fault, we could have more straightforward data flow, by combining queue for instance, rather than having arbitrary recursive flush. WDYT?

Reply:
👍 I removed it for now - let's see when and if we will need it.